### ISSUE TITLE
Suggestion: Hotkey to temporarily prevent direction change.

### ISSUE THREAD

**Issue by GambetTV** [eadorin/about-face#57]
It'd be cool if there was a button you could hold (Ctrl for instance) so that you won't change direction while moving. This way you can simulate backing up without changing directions sometimes!

**Comment by edzillion** [comment]
Good idea, my only issue with this is that CTRL is important for a good few other movement-related modules (for example ShowDragDistance). In fact it's also the hotkey for 'pan canvas' as default. 

Alt is also used for 'Highlight Canvas Objects' but I am not exactly sure that that means.

I wonder would left shift & right shift work?

**Comment by GambetTV** [comment]
Could you just make it configurable? There's like 10,000 modules now, and a lot of them default to the same keys making them difficult to use together! :D

**Comment by LordOHelmet** [comment]
> 
> 
> Added config option to disable automatic rotation on move League-of-Foundry-Developers/about-face#6

Where can one find this option? Is it global or per token?

**Comment by edzillion** [comment]
per token. It's in token settings, on the 'position' tab. 'Disable Direction Indicator'

**Comment by LordOHelmet** [comment]
Thanks for clearing that up!

So unfortunately this is no help to me.

To clear things up. There are two options your module provides:
1. Rotate the token
2. Show a direction indicator
Both can be used induvidually.

Since we use a lot of tokens with no obvious 'front' (left token), rotating them is not an option cause they would sometimes appear to be on their heads.
![token orientation](https://user-images.githubusercontent.com/80178910/117198232-0eaaec80-ade9-11eb-930e-e663e1c9454e.JPG)

That is why we love and need the direction indicator.
To come back to the feature request. It would be nice to have a way to keep the direction indicator, but prevent it from turning while moving by using a hotkey for example.
That way the tokens could side or back step one hex and still keep their orientation (direction indicator).

Do you see any way of implementing this? It would be much appreciated!

**Comment by edzillion** [comment]
Indeed. I will have a look at implementing the hotkey idea in the near future. 

> Since we use a lot of tokens with no obvious 'front' (left token), rotating them is not an option cause they would sometimes appear to be on their heads.

Sorry I wasn't so helpful before. I think you can get what you want by selecting 'Lock Rotation', unselecting 'Disable Direction Indicator', changing 'Flip or Rotate' to Flip Horizontal and then selecting which direction the token is facing by default. Here's what I mean:

![Token Flip Demo](https://raw.githubusercontent.com/League-of-Foundry-Developers/about-face/master/media/token%20flipping%20demo.gif)

**Comment by LordOHelmet** [comment]
Thanks for your detailed reply! I appreciate it very much.
We already use that mechanism and like it.
But what I really am hoping for is this:

![token left](https://user-images.githubusercontent.com/80178910/117206804-94339a00-adf3-11eb-9c21-cd8460e0ae97.JPG)

The token moving to the red hex (left) but keeping his direction indicator facing to the right.
Ending up like this:
![token hex](https://user-images.githubusercontent.com/80178910/117207074-d5c44500-adf3-11eb-88fe-79b6fe5d33b5.JPG)

But using just ONE movement point to go to the left.

**Comment by josmardias** [comment]
So, I would like to use this module but not being able to skip the change facing is kind of painful.

We use vision angle 180º and move points rules, so being able to "step back" without changing facing is important for us. With the "auto face change" of this module the token vision would reveal enemy positioning on it's back.

One idea is a hotkey that prevents this behavior, like suggested by @GambetTV.

Another idea would be to make it configurable what kinds of movement applies the auto facing:
1) moving using the ruler (by pressing space after measuring the distance between the token and another spot)
2) moving using the keyboard
3) moving with mouse drag
However, I don't know enough of FoundryVTT API to say if it's possible to detect token movement type.

In my case I would disable for the mouse drag movement, my players use keyboard and ruler, so the mouse drag would be the escape hatch for a step back.

**Comment by josmardias** [comment]
Maybe the `Misty Step` feature of the `Your Tokens Visible` would work for my case.
I'm assuming their module detects ALT is pressed then sets a new token position without allowing the movement, hence the "misty step" effect.

In that case `About Face` woudn't change the token facing because there is no movement to begin with.

But `Your Tokens Visible` is not currently working with FoundryVTT V9 so I'll have to wait until they upgrade so I can test this hypothesis.

**Comment by josmardias** [comment]
Discussion moved to https://github.com/mclemente/about-face/issues/25
